docs/content/concepts/assets/asset-auto-execution.mdx

```diff
@@ -0,0 +1,169 @@
+
+Setting a lazy auto-materialize policy on an asset allows it to be auto-materialized to help downstream assets meet their freshness policies. In this example, both `asset2` and `asset3` will be auto-materialized up to once per day, to help `asset3` meet its freshness policy.
+
+```python file=/concepts/assets/auto_materialize_lazy_transitive.py
+from dagster import AutoMaterializePolicy, FreshnessPolicy, asset
+
+
+@asset
+def asset1():
+    ...
+
+
+@asset(auto_materialize_policy=AutoMaterializePolicy.lazy())
+def asset2(asset1):
+    ...
+
+
+@asset(
+    auto_materialize_policy=AutoMaterializePolicy.lazy(),
+    freshness_policy=FreshnessPolicy(maximum_lag_minutes=24 * 60),
+)
+def asset3(asset2):
+    ...
+```
+
+### Auto-materialize policies and data versions
+
+If you set an <PyObject object="AutoMaterializePolicy" /> on an asset that's downstream of an [observable source assets](/concepts/assets/asset-observations#observable-source-assets), then changes to the source asset will be treated as new upstream data that can cause the downstream asset to be auto-materialized.
+
+In this example, we check every minute to see whether `source_file` was modified. If it was, then the <PyObject object="AutoMaterializePolicy" /> on `asset1` will cause it to be materialized.
+
+```python file=/concepts/assets/auto_materialize_observable_source_asset.py
+import os
+
+from dagster import (
+    AutoMaterializePolicy,
+    DataVersion,
+    Definitions,
+    ScheduleDefinition,
+    asset,
+    define_asset_job,
+    observable_source_asset,
+)
+
+
+@observable_source_asset
+def source_file():
+    return DataVersion(str(os.path.getmtime("source_file.csv")))
+
+
+@asset(
+    non_argument_deps={"source_file"},
+    auto_materialize_policy=AutoMaterializePolicy.eager(),
+)
+def asset1():
+    ...
+
+
+defs = Definitions(
+    assets=[source_file, asset1],
+    schedules=[
+        ScheduleDefinition(
+            job=define_asset_job(
+                "source_file_observation_job", selection=[source_file]
+            ),
+            cron_schedule="* * * * *",  # every minute
+        )
+    ],
+)
+```
+
+### Auto-materialization and partitions
+
+Partitioned assets can have <PyObject object="AutoMaterializePolicy" pluralize />. Partitions are eligible for auto-materialization when either:
+
+- They are missing - i.e. have never been materialized before
+- New data has arrived in the partitions that are upstream of them
+- They fall within the `time_window_partition_scope_minutes` (more on this below)
+
+Here's a pipeline with two daily-partitioned assets that have eager auto-materialize policies. At the end of each day, a partition for that day will be added to the set of partitions for each of the assets. Dagster will notice that the new partitions exist, but have no materializations, and then auto-materialize them.
```

Comment:
It may also be worth having an example of different time partitions (eg if you have hourly partitions the daemon waits till they are all available before launching the daily partition... link to some thing about partition mapping)

Reply:
I'm going to leave this out for now. I think it would be useful, but doesn't need to be part of the MVP of this guide.

Reply:
and I'm a little worried about overwhelming readers with a lot of examples